File: src/main/java/com/example/solidaryaction/entities/Doacao.java
package com.example.solidaryaction.entities;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Doacao {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;

    public LocalDateTime dataDoacao;

    public BigDecimal valorDoado;

    public String descricao;

    public EnumStatusDoacao status = EnumStatusDoacao.PENDENTE;
}

File: front/app/types/doacao.ts
export class Doacao{
    constructor(
        public id: number | null,
        public dataDoacao:string,
        public valorDoado:number,
        public descricao:string,
        public status:string
    ){}
}
export interface DoacaoFormProps{
    doacaoExistente?:Doacao
}